contracts/core/WaterTowerUpgradeable.sol

```diff
@@ -365,4 +378,17 @@ contract WaterTowerUpgradeable is
     function getTotalRewards() external view returns (uint256 totalRewards) {
         totalRewards = WaterTowerStorage.layout().totalRewards;
     }
+
+    function getAverageStoredWater(address user) external view returns (uint256 amount) {
+        WaterTowerStorage.Layout storage l = WaterTowerStorage.layout();
+        uint256 curPoolIndex = l.curPoolIndex;
+        return l.userPoolHistories[user][curPoolIndex - 1].averageStored;
```

Comment:
This should be checked against curPoolIndex != 0 or else it will revert or?

Reply:
As soon as protocol was deployed, curPoolIndex is increased, so it always greater than 0